src/events/deposits.upsert.service.ts

```diff
@@ -230,172 +232,194 @@ export class DepositsUpsertService {
-              if (!user) {
-                continue;
-              }
+          // Create new deposits
+          await prisma.deposit.createMany({
+            data: depositParams,
+            skipDuplicates: true,
+          });
+
+          deposits = await prisma.deposit.findMany({
+            where: {
+              block_hash: blockHash,
+              network_version: networkVersion,
+            },
+          });
+
+          const eventPayloads = [];
+          const usersFiltered = new Map<string, User>();
+          const eventDepositIds = new Array<number>();
+
+          // Create the event payloads filtering events and users on points < 0
+          for (const deposit of deposits) {
+            const points = this.eventsService.calculateDepositPoints(deposit);
+
+            if (points <= 0) {
+              continue;
+            }
 
-              usersFiltered.set(deposit.graffiti, user);
-              eventDepositIds.push(deposit.id);
+            const user = users.get(deposit.graffiti);
 
-              eventPayloads.push({
-                occurred_at: blockTimestamp.toISOString(),
-                type: EventType.SEND_TRANSACTION,
-                user_id: user.id,
-                points: points,
-                deposit_id: deposit.id,
-              });
+            // This should NEVER happen but can happen if a user is deleted after they made a deposit
+            if (!user) {
+              continue;
             }
 
-            users = usersFiltered;
+            usersFiltered.set(deposit.graffiti, user);
+            eventDepositIds.push(deposit.id);
 
-            // This SHOULD NOT be needed but this cleans up bad data in the server and crashes without this
-            await prisma.event.deleteMany({
-              where: {
-                deposit_id: {
-                  in: eventDepositIds,
-                },
-              },
+            eventPayloads.push({
+              occurred_at: blockTimestamp.toISOString(),
+              type: EventType.SEND_TRANSACTION,
+              user_id: user.id,
+              points: points,
+              deposit_id: deposit.id,
             });
+          }
 
-            await prisma.event.createMany({
-              data: eventPayloads,
-            });
-          } else if (operation.type === BlockOperation.DISCONNECTED) {
-            await prisma.deposit.updateMany({
-              data: {
-                main: false,
-              },
-              where: {
-                block_hash: blockHash,
-                network_version: networkVersion,
-              },
-            });
+          users = usersFiltered;
 
-            deposits = await prisma.deposit.findMany({
-              where: {
-                block_hash: blockHash,
-                network_version: networkVersion,
+          // This SHOULD NOT be needed but this cleans up bad data in the server and crashes without this
```

Comment:
👍

Reply:
Do we care if we lose history of updates here? This would delete the events that were marked `main: false` due to block disconnection, then it would recreate. Seems like the main risk is losing the `created_at` info for previously recorded events, but doesn't seem like a big deal as long as you are fine with it.

Reply:
yeah, that code also hasn't changed it's been in for a week or two in the optimize deposits code. The indent just changed here.